java/vector/src/test/java/org/apache/arrow/vector/ipc/TestArrowStream.java

```diff
@@ -99,4 +98,24 @@ public void testReadWrite() throws IOException {
       }
     }
   }
+
+  @Test
+  public void testReadWriteMultipleBatches() throws IOException {
+    ByteArrayOutputStream os = new ByteArrayOutputStream();
+
+    try (IntVector vector = new IntVector("foo", allocator);) {
+      Schema schema = new Schema(Collections.singletonList(vector.getField()), null);
+      try (VectorSchemaRoot root = new VectorSchemaRoot(schema, Collections.singletonList((FieldVector) vector), vector.getValueCount());
+           ArrowStreamWriter writer = new ArrowStreamWriter(root, new DictionaryProvider.MapDictionaryProvider(), Channels.newChannel(os));) {
```

Comment:
Same thing here, `provider` can just be null and you can just put the output stream directly in the constructor like this and avoid importing Channels:
```
ArrowStreamWriter writer = new ArrowStreamWriter(root, null, out)
```